lua/lspsaga/rename.lua

```diff
@@ -86,6 +86,41 @@ local rename = function()
   apply_action_keys()
 end
 
+--- WorkspaceEdit -> Location[]
+---@param changes table
+---@return table
+local workspaceedit_changes_to_location_list = function(changes)
+  local result = {}
+
+  for uri, text_edits in pairs(changes) do
+    for _, text_edit in ipairs(text_edits) do
+      table.insert(result, {
+        uri = uri,
+        range = text_edit.range
+      })
+    end
+  end
+
+  return result
+end
+
+local rename_handler = function(_, result, ctx, _)
+  -- source: vim.lsp.handlers['textDocument/rename']
```

Comment:
I saw some ppl overwriting lsp handlers directly, and I wonder if this something we should do for all our functions instead of requiring users to set them manually 🤔

Reply:
Sorry if I didn't understand what you were saying.
I think that if you don't specify a handler function, the default function provided by neovim will be called.
https://github.com/neovim/neovim/blob/6f5fae08a302bce6de453425a6b1c1d851112914/runtime/lua/vim/lsp.lua#L1015-L1019

Therefore, specify the handler function only if you want to call your own handler method.

Is this reply correct?

Reply:
Kinda, we are on the same page. see https://github.com/ray-x/navigator.lua/search?q=vim.lsp.handlers. navigator .luaunlike LspSaga doesn't require users to say "map Lspsaga signature_help" but instead it overwrites the vim handler directly so users can just install the plugin and use default vim functions like `vim.lsp.buf.signature_help()` I think.

Reply:
I'm not sure I understand the benefit of changing vim.lsp.handlers.
I don't see a problem with doing it explicitly, like changing the name of Lspsaga.
If vim.lsp.handlers is changed without your knowledge, users will have to change it when they want to use the default handler.

Sorry I can't give a clear reason why one is better than the other.